lib/compat/wordpress-6.9/post-data-block-bindings.php

```diff
@@ -14,13 +14,19 @@
  * @access private
  *
  * @param array    $source_args    Array containing source arguments used to look up the override value.
- *                                 Example: array( "key" => "foo" ).
+ *                                 Example: array( "field" => "foo" ).
  * @param WP_Block $block_instance The block instance.
  * @return mixed The value computed for the source.
  */
 function gutenberg_block_bindings_post_data_get_value( array $source_args, $block_instance ) {
-	if ( empty( $source_args['key'] ) ) {
-		return null;
+	$field = isset( $source_args['field'] ) ? $source_args['field'] : null;
+
+	// Backward compatibility for when the source argument was called `key` in Gutenberg plugin.
+	if ( empty( $field ) && defined( 'IS_GUTENBERG_PLUGIN' ) && IS_GUTENBERG_PLUGIN ) {
+		$field = $source_args['key'];
+		if ( empty( $field ) ) {
+			return null;
+		}
```

Comment:
I see a unit test is failing. It might be related to subtle differences between `isset()` and `empty()`; it's a bit hard to see at a glance if the new code is equivalent to the old code.

How about something like

```suggestion
	if ( empty( $source_args['field'] ) ) {
		// Backward compatibility for when the source argument was called `key` in Gutenberg plugin.
		if ( empty( $source_args['key'] ) ) {
			return null;
		}
		$field = $source_args['key'];
	} else {
		$field = $source_args['field'];
	}	
```

Reply:
The test is till not passing. It's passing on my local though 🤔